command/agent/cache/handler.go

```diff
@@ -20,7 +20,61 @@ import (
 	"github.com/hashicorp/vault/sdk/logical"
 )
 
-func Handler(ctx context.Context, logger hclog.Logger, proxier Proxier, inmemSink sink.Sink, proxyVaultToken bool) http.Handler {
+func ProxyHandler(ctx context.Context, logger hclog.Logger, proxier Proxier) http.Handler {
```

Comment:
It doesn't look like you included auto-auth support here, how come?  I don't see that as intrinsically tied to the cache.

More generally, would it be possible to factor out more of the common elements between the two handlers?  I don't mind if there's some code duplication, but I think the two handlers are going to be extremely similar.

Reply:
The auto-auth token is cached today if `config.Cache.UseAutoAuthToken` is set. I don't think it's a good idea to enable this by default, as it could be considered a security issue (any request to Agent could use its auto-auth token). As a result, I'm more inclined to leave this as something users have to opt in to doing (we don't cache the token unless the config is set).

I can definitely try and refactor the handlers. I'll get on that.

Reply:
I see what you mean - while we do need somewhere to store the token, we don't really need a full cache. I don't really like that we'll be duplicating the config, but I guess that's the price we pay for choosing to implement it this way to begin with. It's doable, like using a new HCL construct like this:

```
api_proxy {
  use_auto_auth_token = true
  force_auto_auth_token = true
}
```

Though this might get confusing as one would have to override the other and we can't deprecate the cache config.

Does this approach make sense?

Reply:
It's worse than it seems actually, because we want all the settings currently in `cache` except for `persist`, so it's not just the auto auth things.

>  we can't deprecate the cache config.

How do you mean?

Reply:
Resolving this (output of this discussion captured in another comment)